zerver/lib/emoji.py

```diff
@@ -149,5 +149,7 @@ def get_emoji_url(emoji_file_name: str, realm_id: int, still: bool = False) -> s
 
 
 def get_emoji_file_name(emoji_file_name: str, emoji_id: int) -> str:
-    _, image_ext = os.path.splitext(emoji_file_name)
+    image_ext = os.path.splitext(emoji_file_name)[1]
+    if not re.match(r"\.\w+$", image_ext):
+        raise JsonableError(_("Bad file name!"))
```

Comment:
What's the background for this change and the `mark_sanitized` one above?

Reply:
We need stricter guarantees -- namely, extensions are how libvips encodes options.  Since we use this extension when saving the thumbnailed emoji, we want to actually remove the untrusted input, locking it down to something reasonable.

Since we are doing actual validation, we no longer need to `mark_sanitized`.